src/wrover_status_light/ServiceIntro.md

```diff
@@ -0,0 +1,160 @@
+```
+#request fields
+string input1
+int32 input2
+---
+#response fields
+float32 output
+```
+After defining a message, ROS would convert the srv file into three classes: service definition, request messages, and response messages. Suppose we have `ExampleService.srv` in an package called `ExamplePackage`, then the names of these packages would be:
+* Service definition: `ExmaplePackage.srv.ExampleService` for Python and `ExamplePackage::ExampleService` for C++
+* Request messages: `ExamplePackage.srv.ExampleServiceRequest` for Python and `ExmaplePackage::ExampleService::Request` for C++
+* Response messages: `ExamplePackage.srv.ExampleServiceResponse` for Python and `ExamplePackage::ExampleService::Request` for C++
+
+## Writing a Server Node
+### Python
+A server code can be initailized with
+```
+def example_service_server():
+    rospy.init_node('example_service_server')
+    s = rospy.Service('example_service', ExamplePackage.srv.ExampleService, example_service_execution)
+    rospy.spin()
+
+def example_service_execution(reqest):
+    return ExamplePackage.srv.ExampleServiceResponse(0.0)
+
+example_service_server()
+```
+To break the code down, the function `example_service_server()` initializes the node.
+```
+rospy.init_node('example_service_server')
+```
+Creates a new node with the name of `example_service_server`.
+```
+rospy.Service('example_service', ExamplePackage.srv.ExampleService, example_service_execution)
+```
+Creates a new service with the name of `example_service` that is defined by the `ExamplePackage.srv.ExampleService` class. When the service is called, the `example_service_execution` method will be called to handle the request. 
+```
+rospy.spin()
+```
+Keeps the code running by entering a loop that listens for incoming requests. 
+
+The function `example_service_execution()` is the handler function for this service and is called whenever the node receives a new request message. Handler functions are responsible for carrying out the actual logics of the service and send a response back to the client. Handler functions always take in a single argument of the request message type, and returns an instance of the response message type.
+
+```
+return ExamplePackage.srv.ExampleServiceResponse(0.0)
+```
+In this example, the handler function always returns a hardcoded response of 0.0. If the handler function wants to use the request inputs, they can be accessed with `request.input1` or `request.input2`.
+
+Returning response messages can also be streamlined by not having to manually create a new instance. In this case, simply 
+```
+return 0.0
+``` 
+would also work. 
+
+### C++
+A server node can be initialized with
+```
+bool ExampleServiceServer::callback(ExmaplePackage::ExampleService::Request& request, ExmaplePackage::ExampleService::Response& response) {
+    return 0.0;
+}
```

Comment:
Why the extra namespace?  This could be a free function, and is probably simpler that way.